### PR TITLE
Remove unused functions in dataset managers

### PR DESCRIPTION
**Pull request by jmchilton**
A problem with this in the mypy upgrade. But I don't think it is used anywhere so I'm opening a separate PR to test this assumption.

## How to test the changes?
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jmchilton** [comment]
Introduced in https://github.com/galaxyproject/galaxy/commit/98db1349a403d46602fe93390501f3703288b639.

**Review by mvdbeek — APPROVED** [review]
🤞

**Comment by nsoranzo** [comment]
It's for sure used in `test/unit/app/managers/test_DatasetManager.py`

File: test/unit/app/managers/test_HistoryManager.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def test_base(self):
             == self.trans.sa_session.execute(select(model.History).filter(model.History.user == user2)).scalar_one()
         )
 
-        history2 = self.history_manager.copy(history1, user=user3)
+        history2 =history1.copy(target_user=user3)
```
Comment (nsoranzo) [inline test/unit/app/managers/test_HistoryManager.py]:
```suggestion
        history2 = history1.copy(target_user=user3)
```

File: test/unit/app/managers/test_DatasetManager.py [hunk 2]
```diff
@@ -33,13 +33,13 @@ def set_up_managers(self):
 
     def test_create(self):
         self.log("should be able to create a new Dataset")
-        dataset1 = self.dataset_manager.create()
+        dataset1 = self.dataset_manager.create_for_unit_testing()
```
Comment (nsoranzo) [inline test/unit/app/managers/test_DatasetManager.py]: I think we need to use `create_for_unit_testing()` only for the 3 tests that were failing: `test_create_public_dataset`, `test_create_private_dataset` and `test_serializers`.

```suggestion
        dataset1 = self.dataset_manager.create()
```

Comment (jmchilton) [reply test/unit/app/managers/test_DatasetManager.py]: I think we want to use the method designed to actually create datasets for all the tests though right? The other one just happens to accidentally work 😅. It shouldn't actually be used I imagine.

Comment (nsoranzo) [reply test/unit/app/managers/test_DatasetManager.py]: I agree on the "It shouldn't actually be used", so what about https://github.com/galaxyproject/galaxy/commit/76aa052ed71fd2e6f454bd4f0c5fa1a527aa392e ?
If you like it, I can push to your branch.

Comment (jmchilton) [reply test/unit/app/managers/test_DatasetManager.py]: Sounds like a good fix - obviously better than leaving unit test fixutres in the main code base! I would say push it.

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**